src/components/interior-left-nav/interior-left-nav.js

```diff
@@ -64,23 +102,94 @@ class InteriorLeftNav extends mixin(createComponent, initComponentBySearch) {
-    if (!('leftNavItemLink' in evt.target.dataset)) {
-      listItem.classList.toggle(this.options.classExpandedLeftNavListItem, !isOpen);
+    const allNestedItems = [
+      ...document.querySelectorAll(
+        this.options.selectorLeftNavListItemHasChildren,
+      ),
+    ];
+    const isOpen = listItem.classList.contains(
+      this.options.classExpandedLeftNavListItem,
+    );
+    allNestedItems.forEach((currentItem) => {
+      if (currentItem !== listItem) {
+        toggleClass(
+          currentItem,
+          this.options.classExpandedLeftNavListItem,
+          false,
+        );
+      }
+    });
+    if (!('InteriorLeftNavItemLink' in evt.target.dataset)) {
+      toggleClass(listItem, this.options.classExpandedLeftNavListItem, !isOpen);
     }
     const list = listItem.querySelector(this.options.selectorLeftNavNestedList);
-    const listItems = [...list.querySelectorAll(this.options.selectorLeftNavNestedListItem)];
+    const listItems = [
+      ...list.querySelectorAll(this.options.selectorLeftNavNestedListItem),
+    ];
     listItems.forEach((item) => {
       if (isOpen) {
         // eslint-disable-next-line no-param-reassign
-        item.querySelector(this.options.selectorLeftNavListItemLink).tabIndex = -1;
+        item.querySelector(
+          this.options.selectorLeftNavListItemLink,
+        ).tabIndex = -1;
       } else {
         // eslint-disable-next-line no-param-reassign
-        item.querySelector(this.options.selectorLeftNavListItemLink).tabIndex = 0;
+        item.querySelector(
+          this.options.selectorLeftNavListItemLink,
+        ).tabIndex = 0;
       }
     });
   }
 
+  toggleLeftNav = () => {
+    const collapsed = this.element.dataset.collapsed === 'true';
+    const eventStart = new CustomEvent(this.options.eventBeforeLeftNavToggled, {
+      bubbles: true,
+      cancelable: true,
+      detail: { collapsed: !collapsed }, // shows where the toggle is going, not where it is
+    });
+
+    if (this.element.dispatchEvent(eventStart)) {
+      if (!collapsed) {
+        this.element.dataset.collapsed = true;
+        this.element.classList.add(this.options.classLeftNavCollapsing);
+
+        window.setTimeout(
+          () => {
+            this.element.classList.remove(this.options.classLeftNavCollapsing);
+            this.element.classList.add(this.options.classLeftNavCollapsed);
+            this.element.dispatchEvent(
+              new CustomEvent(this.options.eventAfterLeftNavToggled, {
+                bubbles: true,
+                cancelable: true,
+                detail: { collapsed: true },
+              }),
+            );
+          },
+          250,
+        );
+      } else {
+        this.element.dataset.collapsed = false;
+        this.element.classList.remove(this.options.classLeftNavCollapsed);
+        this.element.classList.add(this.options.classLeftNavExpanding);
+
+        window.setTimeout(
```

Comment:
The timer should be cancelled when user expands/collapses again. Otherwise the code introduces possibility of race condition (note that the timeout duration is approximate, not guaranteed).